tests/end_to_end/server.py

```diff
@@ -229,3 +245,25 @@ def shutdown(self):
         super(RemoteServerForTesting, self).shutdown()
         if os.path.exists(self.pidFileName):
             os.remove(self.pidFileName)
+
+
+class OidcOpServerForTesting(ServerForTesting):
+    """
+    Runs a test OP server on localhost
+    """
+    def __init__(self):
+        super(OidcOpServerForTesting, self).__init__(
+            oidcOpPort, protocol="https",
+            subdirectory="oidc-provider/simple_op",
+            pingStatusCode=404)
+
+    def start(self):
+        # Setup environment for OP server
+        with tempfile.TemporaryFile() as tf:
```

Comment:
we prefer words as variable names rather than two-letter abbreviations... how about `tempFile` or something?